### PR TITLE
Category search

### PR DESCRIPTION
**Pull request by astrocket**
Done

### CODE REVIEW

**Comment by sydneyitguy** [comment]
#238 #62

**Comment by astrocket** [comment]
image

**Comment by sydneyitguy** [comment]
@cyh76507707 
Here's the draft for the category browsing.
We can probably add some related tags or some top products on the left side.
What do you think?

File: src/Routes.js [hunk 1]
```diff
@@ -23,6 +23,8 @@ const Cookies = asyncComponent(() => import('pages/Cookies'));
 const HuntedList = asyncComponent(() => import('pages/HuntedList'));
 const Profile = asyncComponent(() => import('pages/Profile'));
 const HuntedListByAuthor = asyncComponent(() => import('pages/HuntedListByAuthor'));
+const Category = asyncComponent(() => import('pages/Category'));
```
Comment (sydneyitguy) [inline src/Routes.js]: Let's rename `Category` to `Tag` - sorry that I used a confusing term on the ticket.
(I added reasoning on https://github.com/Steemhunt/api/pull/45#pullrequestreview-138256000)

File: src/features/Post/actions/getPostsByCategory.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    }
+    default:
+      return state;
+  }
+}
+
+/*--------- SAGAS ---------*/
+function* getPostsByCategory({ category, page }) {
+  try {
+    const result = yield api.get(`/category/${category}.json`, { page: page, sort: getSortOption('profile') });
```
Comment (sydneyitguy) [inline src/features/Post/actions/getPostsByCategory.js]: We save sort options differently on each list
`getSortOption('profile') => getSortOption('tag')`

File: src/features/Post/actions/setCurrentCategory.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import { put, select, takeLatest } from 'redux-saga/effects';
+import update from 'immutability-helper';
+
```
Comment (sydneyitguy) [inline src/features/Post/actions/setCurrentCategory.js]: Do we actually need this actions at all?
I know I probably have the same thing for `author`.. maybe that's also a mistake,
but that `currentCategory` is on URL and can be accessed via  `match.params.category` on the component anyway.

So whey we need `setCurrentCategory` from the first place? lol

File: src/Routes.js (hunk 1 above)
Comment (astrocket) [reply src/Routes.js]: Sure I agree with that. Actually, for some reason I realized that the front-end terminology does not match the backend record name almost at the end of the branch work. Let me fix all on friday morning.

File: src/features/Post/actions/getPostsByCategory.js (hunk 2 above)
Comment (astrocket) [reply src/features/Post/actions/getPostsByCategory.js]: Checked

File: src/features/Post/actions/setCurrentCategory.js (hunk 3 above)
Comment (astrocket) [reply src/features/Post/actions/setCurrentCategory.js]: You are right. I thought it was a little bit weird too. Let's discard those.

**Comment by sydneyitguy** [comment]
potential stuff on left panel:
- related tags
- top products

**Comment by astrocket** [comment]
image

**Comment by sydneyitguy** [comment]
@cyh76507707 also for the search feature,
that may require some additional DB tables and API structures..
kind of too much dev effort & adding complexity for a little features.

I guess the current version @astrocket posted above would be fine with a little design tuning..
what do you think?

**Comment by sydneyitguy** [comment]
image

**Comment by cyh76507707** [comment]
category
1. 프로필 사진은 저 태그 가장 최근 포스트의 스샷
2. 태그 마우스 오버시/클릭시 색깔: #FFEBEB

**Pull request merged**